lib/card_generator.rb

```diff
@@ -0,0 +1,47 @@
+require 'json'
+
+class CardGenerator
+  attr_reader :cards, :loaded_deck, :card_inputs
+  def initialize(cards = nil)
```

Comment:
Your solution for iteration for doesn't quite follow the interaction pattern exactly.

The pattern looks like so:

```ruby
pry(main)> require './lib/card_generator'
#=> true

pry(main)> filename = "cards.txt"
#=> "cards.txt"

pry(main)> cards = CardGenerator.new(filename).cards
#=> [#<Card:0x007f9f1413cbe8...>, ... #<Card:0x007f9f1413c2b0...>]
```

I am supposed to be able to create a new instance of `CardGenerator` and pass it the filename that I want it to read and it should return an array of cards. The way your `CardGenerator` class is set up, it is expecting an array of cards passed in when you go to create an instance of `CardGenerator`, which is not expected behavior.